lib/iteration.rb

```diff
@@ -0,0 +1,109 @@
+def benchmark(num = 1)
+  start = Time.now
+  num.times { yield }
+  elapsed_time = (Time.now - start) / num
+end
+
+def race(procs, num = 1)
+  arr = []
+  procs.each do |proc|
+    runtime = benchmark(num)
+    arr << runtime
+  end
+  index = arr.index(arr.max)
+  procs[index]
+end
+
+def eval_block(*args, &prc)
+  if prc.nil?
+    "NO BLOCK GIVEN"
+  else
+    p yield *args
+  end
+end
+
+def factors(num)
+  (1..num).select { |n| num % n == 0 }
+end
+
+class Array
+  def bubble_sort!(&prc)
```

Comment:
Yes. You sort by comparing the current element to the one after, and keep going until the end of the array. Then you repeat. The largest number would naturally get sorted to the end the first round.